zkvm/src/merkle.rs

```diff
@@ -22,7 +22,7 @@ pub enum MerkleNeighbor {
 pub struct MerkleTree {
     size: usize,
     label: &'static [u8],
-    root: MerkleNode,
+    root: Option<MerkleNode>,
```

Comment:
Maybe use MerkleNode::Leaf([u8; 32]) and avoid optional here? Or add ::Empty variant if that’d be cleaner.